docs/HowToSupportTizenRtProtectedBuild.md

```diff
@@ -0,0 +1,343 @@
+
+#### Defconfig
+- The following configs need to be set or unset in the defconfig file.
+
+```
+CONFIG_BUILD_FLAT is not set
+CONFIG_BUILD_PROTECTED=y
+CONFIG_BUILD_2PASS=y
+CONFIG_PASS1_TARGET="all"
+CONFIG_PASS1_OBJECT=""
+CONFIG_TINYARA_USERSPACE=0x08080000
+CONFIG_ARM_HAVE_MPU_UNIFIED=y
+CONFIG_ARMV7M_MPU=y
+CONFIG_ARMV7M_MPU_NREGIONS=8
+CONFIG_MM_KERNEL_HEAP=y
+CONFIG_LIB_SYSCALL=y
+CONFIG_SYS_RESERVED=8
+CONFIG_SYS_NNEST=2
+```
+- Here CONFIG_TINYARA_USERSPACE should provide the start address of the user binary in the flash memory. This address depends on how the flash is partitioned into kernel and user areas. This config needs to be updated whenever there is a change in the start address of the user binary.
+
+#### Make.defs
+- The make.defs file provides several definitions which are used during the build process. One of the definitions is (ARCHSCRIPT) for the linker script to be used to link the kernel binary.
+- In case of flat build, a single linker script such as ld.script is used. The same is specified in the make.defs file.
+- For the case of protected build, we need to provide two linker scripts:
+	1. memory.ld : Provides the overall memory layout
+	2. kernel-space.ld : Provides the rules to link the kernel binary
+- The procedure to create these two linker scripts is provided in the following sections.
+- In order to include the definitions of these linker scripts, the make.defs should be modified as follows:
+
+```
+MEM_LDSCRIPT = memory.ld
+KSPACE_LDSCRIPT = kernel-space.ld
+....
+ARCHSCRIPT = -T$(TOPDIR)/../build/configs/$(CONFIG_ARCH_BOARD)/scripts/$(MEM_LDSCRIPT)
+ARCHSCRIPT += -T$(TOPDIR)/../build/configs/$(CONFIG_ARCH_BOARD)/scripts/$(KSPACE_LDSCRIPT)
+```
+
+
+### Creating Linker Scripts
+- In case of flat build, the single linker script ld.script describes the location and size of memory regions in the target, and also describes the placement and linking of different sections of binary into the different memory regions. In case of protected build three linker scripts are needed to perform this functionality:
+	- memory.ld : This describes the location and size of memory regions in the target.
+	- kernel-space.ld : This is used to assign sections of kernel binary to particular memory regions.
+	- user-space.ld : This is used to assign sections of user binary to particular memory regions.
+- The kernel-space.ld and user-space.ld make use of the memory regions defined in memory.ld.
+- All these three files should be created for protected build and placed in build/configs/<CHIPSET>/scripts folder.
+- The following sections give the procedure to create these linker files.
+
+#### Memory.ld
+- This script defines the code and data memory regions to be used by the kernel and user linker scripts.
+- Generally a board will contain two types of memory. First is the flash memory which will contain the code and second is the RAM memory to be used for data, stack and heap during execution
```

Comment:
1. Could you let me know why you add this line? Especially `which will contain the code` and `to be used for data, stack and heap during execution`? How about non-XIP like Mediatek board?
2. Please add the dot at end of sentence.

Reply:
We added some information about the non-XIP case in the new document.